packages/Python/lldbsuite/test/lang/swift/implementation_only_imports/main_executable/TestMainExecutable.py

```diff
@@ -0,0 +1,147 @@
+
+class TestMainExecutable(TestBase):
+
+    mydir = TestBase.compute_mydir(__file__)
+
+    def launch_info(self):
+        info = lldb.SBLaunchInfo([])
+
+        if self.getPlatform() == "freebsd" or self.getPlatform() == "linux":
+            # LD_LIBRARY_PATH must be set so the shared libraries are found on
+            # startup
+            library_path = os.environ.get("LD_LIBRARY_PATH", "")
+            if library_path:
+                library_path += ":"
+            library_path += self.getBuildDir()
+
+            info.SetEnvironmentEntries(["LD_LIBRARY_PATH=" + library_path], True)
+
+        return info
+
+    @swiftTest
+    def test_implementation_only_import_main_executable(self):
+        """Test `@_implementationOnly import` in the main executable
+
+        See the ReadMe.md in the parent directory for more information.
+        """
+
+        self.build()
+        def cleanup():
+            lldbutil.execute_command("make cleanup")
+        self.addTearDownHook(cleanup)
+        lldbutil.run_to_source_breakpoint(self, "break here", lldb.SBFileSpec("main.swift"), self.launch_info())
+
+        # This test is deliberately checking what the user will see, rather than
+        # the structure provided by the Python API, in order to test the recovery.
+        self.expect("fr var", substrs=[
+            "(SomeLibrary.TwoInts) value = (first = 2, second = 3)",
+            "(main.ContainsTwoInts) container = {\n  wrapped = (first = 2, second = 3)\n  other = 10\n}",
+            "(Int) simple = 1"])
+        self.expect("e value", substrs=["(SomeLibrary.TwoInts)", "= (first = 2, second = 3)"])
+        self.expect("e container", substrs=["(main.ContainsTwoInts)", "wrapped = (first = 2, second = 3)", "other = 10"])
+        self.expect("e TwoInts(4, 5)", substrs=["(SomeLibrary.TwoInts)", "= (first = 4, second = 5)"])
+    
+    @swiftTest
+    def test_implementation_only_import_main_executable_no_library_module(self):
+        """Test `@_implementationOnly import` in the main executable, after removing the library's swiftmodule
+        
+        See the ReadMe.md in the parent directory for more information.
+        """
+
+        self.build()
+        os.remove(self.getBuildArtifact("SomeLibrary.swiftmodule"))
+        os.remove(self.getBuildArtifact("SomeLibrary.swiftinterface"))
+        def cleanup():
+            lldbutil.execute_command("make cleanup")
+        self.addTearDownHook(cleanup)
+        lldbutil.run_to_source_breakpoint(self, "break here", lldb.SBFileSpec("main.swift"), self.launch_info())
+
+        # FIXME: This particular test config is producing different results on
+        # different machines, but it's also the least important configuration
+        # (trying to debug something built against a library without that
+        # library present, implementation-only or not, and that library doesn't
+        # even have library evolution support enabled. Just make sure we don't
+        # crash.
```

Comment:
@adrian-prantl @dcci Is this acceptable? I don't like it but I've already sunk way too much time into these few tests.

Reply:
They produce different results, but they are all different kinds of failures?
We *are* checking for the existence of any error, so I guess that is okay.

Reply:
They are failing in different ways and it's really rather bizarre to me. I could file a Radar or Jira if you want it explicitly tracked somehow, but as noted it's not the most interesting configuration and I don't expect anyone to spend time tracking it down in practice.

Reply:
As long as we are testing for the fact *that they fail* that is good enough at the moment.